cmd/modelcmd/controller.go

```diff
@@ -287,8 +289,22 @@ func (w *sysCommandWrapper) Init(args []string) error {
 	}
 	if w.controllerName != "" {
 		if err := w.SetControllerName(w.controllerName); err != nil {
-			return errors.Trace(err)
+			return translateControllerError(w.ClientStore(), err)
 		}
 	}
-	return w.ControllerCommand.Init(args)
+	return w.ControllerCommand.Run(ctx)
+}
+
+func translateControllerError(store jujuclient.ClientStore, err error) error {
+	if !errors.IsNotFound(err) {
+		return err
+	}
+	controllers, err := store.AllControllers()
+	if err != nil {
+		return err
+	}
+	if len(controllers) == 0 {
+		return ErrNoControllersDefined
+	}
+	return ErrNoCurrentController
 }
```

Comment:
New code is here: https://github.com/juju/juju/pull/6306/commits/093452082582554c41437c29eb485de248721237#diff-3475f343e517b41c7eb532ace14c4655R298